BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs

```diff
@@ -0,0 +1,176 @@
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(List<MarkDownModel>))]
+        public async Task<IActionResult> GetAll()
+        {
+            try
+            {
+
+                var documents = await markDownService.GetAllDocuments();
+
+                var markdowns = documents?.Select(d => new MarkDownModel
+                {
+                    Id = d.Id,
+                    Body = d.MarkDown,
+                }).ToList();
+
+
+
+                return Ok(markdowns);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
+            }
+        }
+
+        [Authorize]
+        [HttpPost]
+        [Route("api/markdown/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Post([FromBody] MarkDownModel markDownModel)
+        {
+            if (string.IsNullOrWhiteSpace(markDownModel?.Body))
+            {
+                return BadRequest("model is not OK");
+            }
+
+            try
+            {
+                var updatedDocument = await markDownService.UpdateDocument(markDownModel.Id, markDownModel.Body);
+
+                if (!string.IsNullOrWhiteSpace(updatedDocument?.MarkDown))
+                {
+                    var updatedMarkDownModel = new MarkDownModel
+                    {
+                        Id= updatedDocument.Id,
+                        Body = updatedDocument.MarkDown,
+                    };
+                    return Ok(updatedMarkDownModel);
+                }
+                else
+                {
+                    throw new Exception("Unable to retrieve updated document");
+                }
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
+            }
+        }
+
+        [Authorize]
+        [HttpPut]
+        [Route("api/markdown/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Put([FromBody] MarkDownModel markDownModel)
+        {
+            if (string.IsNullOrWhiteSpace(markDownModel?.Body))
+            {
+                return BadRequest("model is not OK");
+            }
+
+            try
+            {
+                var insertedDocument = await markDownService.InsertDocument(markDownModel.Body);
+
+                if (!string.IsNullOrWhiteSpace(insertedDocument?.MarkDown))
+                {
+                    var insertedMarkDownModel = new MarkDownModel
+                    {
+                        Id = insertedDocument.Id,
+                        Body = insertedDocument.MarkDown,
+                    };
+                    return Ok(insertedMarkDownModel);
+                }
+                else
+                {
+                    throw new Exception("Unable to retrieve inserted document");

```

Comment:
I don't understand the pattern here, let talk about it in details

Reply:
Yes for this part really my bad, quite disapointed from my side about this beginner bad practice. 

I fixed it on https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e